test/robust.jl

```diff
@@ -0,0 +1,29 @@
+using StatsBase
+using Base.Test
+
+### Trimming outliers
+
+@test trim([1,2,3,4,5,6,7,8])           == [2,3,4,5,6,7]
+@test trim([1,2,3,4,5,6,7,8], 0.1)      == [1,2,3,4,5,6,7,8]
+@test trim([1,2,3,4,5,6,7,8], 0.2)      == [2,3,4,5,6,7]
+@test trim([1,2,3,4,5,6,7,8,9], 0.4)    == [4,5,6]
+
+@test trim!([1,2,3,4,5,6,7,8])          == [2,3,4,5,6,7]
+@test trim!([1,2,3,4,5,6,7,8], 0.1)     == [1,2,3,4,5,6,7,8]
+@test trim!([1,2,3,4,5,6,7,8], 0.2)     == [2,3,4,5,6,7]
+@test trim!([1,2,3,4,5,6,7,8,9], 0.4)   == [4,5,6]
+
+@test winsor([1,2,3,4,5,6,7,8])         == [2,2,3,4,5,6,7,7]
+@test winsor([1,2,3,4,5,6,7,8], 0.1)    == [1,2,3,4,5,6,7,8]
+@test winsor([1,2,3,4,5,6,7,8], 0.2)    == [2,2,3,4,5,6,7,7]
+@test winsor([1,2,3,4,5,6,7,8,9], 0.4)  == [4,4,4,4,5,6,6,6,6]
+
+@test winsor!([1,2,3,4,5,6,7,8])        == [2,2,3,4,5,6,7,7]
+@test winsor!([1,2,3,4,5,6,7,8], 0.1)   == [1,2,3,4,5,6,7,8]
+@test winsor!([1,2,3,4,5,6,7,8], 0.2)   == [2,2,3,4,5,6,7,7]
+@test winsor!([1,2,3,4,5,6,7,8,9], 0.4) == [4,4,4,4,5,6,6,6,6]
+
```

Comment:
Should probably add tests that trigger the error handling in your function. For that you can use `@test_throws`.